### DISCUSSION TITLE
Created and updated fields are empty in the create event for collections with "autodate" fields.

### DISCUSSION THREAD

**Discussion by EssekerDev** [pocketbase/pocketbase#5973] [Q&A] [answered]
### Description
When adding an entry to a PocketBase collection configured with automatic `created` and `updated` fields, the create event triggered by an active subscription does not contain values for these fields. 
This seems to happen because the event is sent before PocketBase finishes populating the created and updated fields.
Am I missing something in the setup or configuration of the collection that could explain this behavior?

### Technology
- Flutter Application
- Dart SDK for PocketBase version 0.19.1
- PocketBase version 23.2 (Docker)
- Client Platform: Flutter on Windows.

### Steps to Reproduce
1. Set up a PocketBase collection with `created` and `updated` fields configured as auto-generated by the database.
2. Create an active subscription to the collection using the Dart SDK.
3. Add a new entry to the collection via the API or another method.
4. Observe the create event received on the client via the subscription.

**Expected Result**
The create event should include the `created` and `updated` fields with their correct values (auto-generated timestamps).

**Observed Result**
The `created` and `updated` fields are empty ("") in the create event.
However, these fields are correctly populated when the database confirms the model after the initial addition.

### Impact
Clients relying on create events cannot use the `created` and `updated` fields reliably.
Requires additional fetches for complete data, leading to extra API calls and reduced performance.

### Hypothesis
The create event appears to be triggered before PocketBase finishes auto-generating the `created` and `updated` fields.

### Example of Observed Data
- Create event received:
```
{
  "collectionId": "jt66zfogdlmjt70",
  "collectionName": "vehicles",
  "created": "",
  "id": "ju3j193488d0jtj",
  "updated": "",
  "vehicleBrand": "Exemple",
  "vehicleColor": "Violet"
}
```
- Data confirmed after the API addition:
```
{
  "collectionId": "jt66zfogdlmjt70",
  "collectionName": "vehicles",
  "created": "2024-11-28T19:53:39.739Z",
  "id": "ju3j193488d0jtj",
  "updated": "2024-11-28T19:53:39.739Z",
  "vehicleBrand": "Exemple",
  "vehicleColor": "Violet"
}
```

**Comment by ganigeorgiev** [comment]
Before broadcasting the record we create a shallow copy of the updated model and this seems like a bug with the record `Fresh()/Clone()` methods caused by the fact that the `autodate` field doesn't have setters and internally instead of `record.Set(k, v)` probably we will have to replace it with `record.SetRaw(k, v)` when cloning the fields. I'll investigate it in more details tomorrow.

**Reply by EssekerDev** [reply]
Thanks for the clarification! I'll implement a workaround for now by fetching the complete record after receiving the event. I'll also keep an eye on future updates for the fix. Appreciate your support!

**Reply by ganigeorgiev** [reply, marked as answer]
I've pushed a small fix in the [`develop`](https://github.com/pocketbase/pocketbase/tree/develop).

Sometime tomorrow I'll update the tests and will tag a minor release. If you don't want to wait, you can build the `develop` branch on your own:

0. _[Install Go 1.23+](https://go.dev/doc/install) (if you haven't already)_
1. Clone the `develop` branch:
   ```sh
   git clone -b develop --single-branch https://github.com/pocketbase/pocketbase.git
   ```
2. Navigate to `examples/base`
2. Run `CGO_ENABLED=0 go build`
3. Start the created executable by running `./base serve`.

**Reply by EssekerDev** [reply]
Thank you so much for your quick response and for addressing the issue so promptly! Your commitment and engagement with the community are truly appreciated. I'll try building the develop branch as you suggested, and I'm looking forward to the upcoming minor release. Thanks again for the excellent support and dedication to making PocketBase even better! 🙌

Edit : 
I built the develop branch as suggested, and the fix works perfectly! The created and updated fields are now correctly included in the event. Thank you once again for the swift resolution! 🚀
